pkgs/applications/audio/emacspeak/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, fetchgit, rsync, gnused, emacs, tcl, espeak-classic }:
+
+stdenv.mkDerivation rec {
+  pname = "emacspeak";
+  version = "50.0";
+
+  src = fetchgit {
+    url = "https://github.com/tvraman/emacspeak";
+    rev = version;
+    sha256 = "043mn3rr48gx2mi4rnkcvp09fwhs88yyqcszavkzvvd57mh4vd0h";
+  };
+
+  nativeBuildInputs = [rsync gnused] ;
+  buildInputs = [ emacs tcl espeak-classic ];
+  postPatch = ''
+    substituteInPlace servers/native-espeak/Makefile --replace '/usr/include/tcl$(TCL_VERSION)' ${tcl}/include
+    substituteInPlace servers/native-espeak/Makefile --replace 'LIBPARENTDIR = ''${PREFIX}/'  'LIBPARENTDIR = /'
+'';
+
+  preConfigure = ''
+    make config
+  '';
+
+  postBuild = ''
+  make espeak
+'';
+
+# installPhase = ''
+#   mkdir -p $out/share/emacs/site-lisp/emacspeak/servers/native-espeak
+# '';
+postInstall = ''
+  cd servers/native-espeak
+  make DESTDIR=$out install
+  cd ../../
```

Comment:
This seems odd. Do you have to call a specific Makefile?
```
makeFlags = [ "DESTDIR=${placeholder "out"}" ];
```

Reply:
Yeah, there's a  separate Makefile that builds `tclspeak.so`. Main Makefile doesn't do install. It just prints message to add current folder to emacs config. Could you explain a bit about `placeholder`? `$out` is deprecated?

Reply:
in most situations, there's no difference. From what I understand, placeholder just ensures that a variable doesn't get resolved until the last possible second. I don't remember the exact scenarios in which it was necessary to add the placeholder feature tbh

Reply:
but these three lines could be condensed to:
```
make -C servers/native-espeak DESTDIR=$out install
```